### PR TITLE
Navneet sahota/feat#81

### LINKED ISSUE #81: Create Add post component

### PR DESCRIPTION
**Pull request by Navneet-Sahota**
close #81 #98 

## Types of changes

- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Navneet-Sahota** [comment]
Add_post

**Comment by Navneet-Sahota** [comment]
In this PR, I have implemented a route called `/new` where I have implemented two components
1. WritePost - where the user will write down his markdown.
2. PreviewPost - where the user can preview his blog.

For **WritePost** I used `react-autosize-textarea` for resizable textarea.
For **PreviewPost** I used `react-markdown` to support markdown conversion and added `remark-iframes` to it to support embedded videos.

There's also a **Help** tab where we can provide basic Markdown syntax to the user.

**Comment by Navneet-Sahota** [comment]
Currently, this PR does not have tests. I'm writing test right now. Meanwhile, you guys can review the rest of it.

File: client/src/API/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const createPost = async (body) => {
+  let response;
+  try {
+    response = await axios.post('posts', body, axiosConfig);
+    if (!response.data.success) {
+      throw new Error(response.data.message);
+    }
+  } catch (err) {
+    console.error(err);
```
Comment (sumeetrohra) [inline client/src/API/index.js]: Remove consoles and show appropriate error messages on screen

File: client/src/API/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          key: categoryData._id,
+          value: categoryData._id,
+          text: categoryData.name,
+        }
+      ));
+    } else {
+      throw new Error(response.data.message);
+    }
+  } catch (err) {
+    console.error(err);
```
Comment (sumeetrohra) [inline client/src/API/index.js]: same

File: client/src/API/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  return response;
+};
+
+const getCategories = async () => {
+  let response;
+  try {
+    response = await axios.get('categories', axiosConfig);
+    if (response.data.success) {
+      response = response.data.categories.map((categoryData) => (
```
Comment (sumeetrohra) [inline client/src/API/index.js]: Don't overwrite the response object

File: client/src/API/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    console.error(err);
+  }
+  return response;
+};
+
+const getCategories = async () => {
+  let response;
+  try {
+    response = await axios.get('categories', axiosConfig);
+    if (response.data.success) {
```
Comment (rishabh1403) [inline client/src/API/index.js]: if(response && response.data && response.data.success )

File: client/src/Routes.jsx [hunk 5]
```diff
@@ -5,6 +5,7 @@ import {
 
 import Login from './views/Login';
 import Signup from './views/Signup';
+import New from './views/New';
```
Comment (rishabh1403) [inline client/src/Routes.jsx]: can we name it better ?

File: client/src/components/WritePost/index.jsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  const handleSave = async () => {
+    setIsSaveLoading(true);
+    const body = {
+      author: '5d5d60a51930b06b21e65765',
+      title,
+      content,
+      public: isPublic,
+      published: false,
+    };
+    await createPost(body);
```
Comment (rishabh1403) [inline client/src/components/WritePost/index.jsx]: try/catch ?

File: client/src/API/index.js (hunk 1 above)
Comment (Navneet-Sahota) [reply client/src/API/index.js]: Placeholders for now. Will replace later

File: client/src/API/index.js (hunk 3 above)
Comment (Navneet-Sahota) [reply client/src/API/index.js]: Will change it

File: client/src/Routes.jsx (hunk 5 above)
Comment (Navneet-Sahota) [reply client/src/Routes.jsx]: Here I'm following the convention of naming components after the route they render in, which in this case is `/new`. But agree that it's not a good component name.
`NewPost` was already taken.

File: client/src/components/WritePost/index.jsx (hunk 6 above)
Comment (Navneet-Sahota) [reply client/src/components/WritePost/index.jsx]: `createPost` have a try/catch inside it. Should I wrap function call in a different try/catch as well?

**Pull request merged**